### PR TITLE
Added initial ACME application

### PR DESCRIPTION
**Pull request by edewata**
This patch added the initial ACMEApplication and CLIs to deploy
and undeploy the application. Other functionalities will be added
in subsequent patches.

### CODE REVIEW

File: base/server/share/conf/pki.policy [hunk 1]
```diff
@@ -13,21 +13,11 @@
...
 
-        // Allow Tomcat JULI to read subsystem logging configuration.
-        permission java.io.FilePermission "/usr/share/pki/server/webapps/pki/WEB-INF/classes/logging.properties", "read";
-        permission java.io.FilePermission "/usr/share/pki/ca/webapps/ca/WEB-INF/classes/logging.properties", "read";
-        permission java.io.FilePermission "/usr/share/pki/kra/webapps/kra/WEB-INF/classes/logging.properties", "read";
-        permission java.io.FilePermission "/usr/share/pki/ocsp/webapps/ocsp/WEB-INF/classes/logging.properties", "read";
-        permission java.io.FilePermission "/usr/share/pki/tks/webapps/tks/WEB-INF/classes/logging.properties", "read";
-        permission java.io.FilePermission "/usr/share/pki/tps/webapps/tps/WEB-INF/classes/logging.properties", "read";
+        // Allow Tomcat JULI to read subsystem logging.properties files.
+        permission java.io.FilePermission "/usr/share/pki/-", "read";
```
Comment (cipherboy) [inline base/server/share/conf/pki.policy]: I'm not an expert here. But won't this give the subsystems access to far more than "just" `logging.properties` files?

File: base/server/python/pki/server/cli/acme.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        descriptor = os.path.join(pki.server.PKIServer.SHARE_DIR,
+                                  'acme/conf/Catalina/localhost/acme.xml')
+        doc_base = os.path.join(pki.server.PKIServer.SHARE_DIR,
+                                'acme/webapps/acme')
+
+        logging.info('Deploying %s webapp', name)
+        instance.deploy_webapp(name, descriptor, doc_base)
+
+
+class ACMEUndeployCLI(pki.cli.CLI):
```
Comment (cipherboy) [inline base/server/python/pki/server/cli/acme.py]: Just gotta say, "undeploy" sounds funny but I guess its standard terminology so I'm fine with it. :)

File: base/server/share/conf/pki.policy (hunk 1 above)
Comment (edewata) [reply base/server/share/conf/pki.policy]: Yes, but I don't see a problem with that. Those are shared files owned by the package and they are supposed to be read-only.

File: base/server/python/pki/server/cli/acme.py (hunk 2 above)
Comment (edewata) [reply base/server/python/pki/server/cli/acme.py]: Yeah, it seems to be the commonly used term, but I'm open to suggestions.

File: base/server/share/conf/pki.policy (hunk 1 above)
Comment (cipherboy) [reply base/server/share/conf/pki.policy]: Hmmm, I guess that's fair. We generally assume that any local process should be able to read those files.

Just that the comment seems misleading if we're granting access to everything. :)

**Comment by edewata** [comment]
The comment in pki.policy is fixed as discussed. Thanks!

**Pull request merged**